### ISSUE TITLE
Prometheus Metrics Improvements

### ISSUE THREAD

**Issue by jonrandahl** [epimorphics/hmlr-linked-data#148]
Improve the metrics endpoint on all HMLR applications to include the `process_threads` metric, labelled by status for additional filtering and sorting via the integrated metric dashboards.

> ```ruby
>     # description: 'Thread is aborting'
>     Prometheus::Client.registry
>                       .get(:process_threads)
>                       .set(
>                         Thread.list.select { |thread| thread.status == 'aborting' }.count,
>                         labels: {
>                           status: 'aborting'
>                         }
>                       )
>     # description: 'Thread is sleeping or waiting on I/O'
>     Prometheus::Client.registry
>                       .get(:process_threads)
>                       .set(
>                         Thread.list.select { |thread| thread.status == 'sleep' }.count,
>                         labels: {
>                           status: 'sleep'
>                         }
>                       )
>     # description: 'Thread is executing'
>     Prometheus::Client.registry
>                       .get(:process_threads)
>                       .set(
>                         Thread.list.select { |thread| thread.status == 'run' }.count,
>                         labels: {
>                           status: 'run'
>                         }
>                       )
>     # description: 'Thread is terminated normally'
>     Prometheus::Client.registry
>                       .get(:process_threads)
>                       .set(
>                         Thread.list.select { |thread| thread.status == false }.count,
>                         labels: {
>                           status: 'false'
>                         }
>                       )
>     # description: 'Thread is terminated with an exception'
>     Prometheus::Client.registry
>                       .get(:process_threads)
>                       .set(
>                         Thread.list.select { |thread| thread.status.nil? }.count,
>                         labels: {
>                           status: 'nil'
>                         }
>                       )
> ```

**Comment by tomguilbert** [comment]
@jonrandahl can this be closed now?

**Comment by jonrandahl** [comment]
Closing in favour of updated tickets on individual application repos
